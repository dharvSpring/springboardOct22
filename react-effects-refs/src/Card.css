.Card {
    position: absolute;
    margin-top: 10rem;
    margin-left: -200px;
}

.rotate-5 {
    transform: rotate(5deg);
}

.rotate-10 {
    transform: rotate(10deg);
}

.rotate-15 {
    transform: rotate(15deg);
}

.rotate-20 {
    transform: rotate(20deg);
}

.rotate-25 {
    transform: rotate(25deg);
}

.rotate-355 {
    transform: rotate(355deg);
}

.rotate-350 {
    transform: rotate(350deg);
}

.rotate-345 {
    transform: rotate(345deg);
}

.rotate-340 {
    transform: rotate(340deg);
}

.rotate-335 {
    transform: rotate(335deg);
}