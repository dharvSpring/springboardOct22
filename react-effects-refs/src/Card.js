import React from "react";
import './Card.css';

function Card({rotClass, imgUrl}) {
    const cssClasses = `Card ${rotClass}`;
    return (
        <img className={cssClasses} src={imgUrl} />
    )
}

export default Card;