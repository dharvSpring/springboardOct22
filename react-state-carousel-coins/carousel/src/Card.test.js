import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Card from "./Card";

it("card smoketest no props", function() {
    render(<Card />);
});

it("card smoketest w/ props", function() {
    render(<Card caption="test" src="text" currNum="1" totalNum="1" />);
});

it("card snapshot w/ props", function() {
    const { asFragment } = render(<Card caption="test" src="text" currNum="1" totalNum="1" />);
    expect(asFragment()).toMatchSnapshot();
});