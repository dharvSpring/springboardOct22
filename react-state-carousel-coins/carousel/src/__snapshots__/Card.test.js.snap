// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card snapshot w/ props 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      test
    </h4>
    <img
      alt="test"
      class="Card-image"
      src="text"
    />
    <small
      class="Card-small"
    >
      Image 1 of 1.
    </small>
  </div>
</DocumentFragment>
`;
